ui-manchette/src/stories/Manchette.stories.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useRef, useState } from 'react';
+
+import '@osrd-project/ui-core/dist/theme.css';
+import '@osrd-project/ui-manchette/dist/theme.css';
+import { EyeClosed, Telescope } from '@osrd-project/ui-icons';
+import type { Meta, StoryObj } from '@storybook/react';
+
+import { SAMPLE_PATH_PROPERTIES_DATA } from './assets/sampleData';
+import Menu, { type MenuItem } from './components/Menu';
+import Manchette from '../components/Manchette';
+import { type StyledOperationalPointType } from '../types';
+
+const OperationalPointListData: StyledOperationalPointType[] =
+  SAMPLE_PATH_PROPERTIES_DATA.operational_points?.map((op) => ({ ...op, display: true })) ?? [];
+
+const meta: Meta<typeof Manchette> = {
+  component: Manchette,
+  title: 'Manchette/Manchette',
+  tags: ['autodocs'],
+  argTypes: {
+    operationalPoints: {
+      control: {
+        type: 'object',
+      },
+    },
+    zoomYIn: {
+      action: 'zoomYIn',
+    },
+    zoomYOut: {
+      action: 'zoomYOut',
+    },
+  },
+};
+
+const ManchetteWithWaypointMenu = () => {
+  const [menuPosition, setMenuPosition] = useState<{ top: number; left: number } | null>(null);
+  const [activeOperationalPointId, setActiveOperationalPointId] = useState<string>();
+
+  const menuRef = useRef<HTMLDivElement>(null);
+
+  const menuItems: MenuItem[] = [
+    {
+      title: 'Action 1',
+      icon: <EyeClosed />,
+      onClick: () => {
+        setMenuPosition(null);
+        setActiveOperationalPointId(undefined);
+      },
+    },
+    {
+      title: 'Action 2',
+      icon: <Telescope />,
+      onClick: () => {
+        setMenuPosition(null);
+        setActiveOperationalPointId(undefined);
+      },
+    },
+  ];
+
+  const handleWaypointClick = (id: string, ref: HTMLDivElement | null) => {
+    if (!ref) return;
+    const position = ref.getBoundingClientRect();
+    setMenuPosition({ top: position.bottom - 2, left: position.left });
+    setActiveOperationalPointId(id);
+  };
+
+  return (
+    <Manchette
+      operationalPoints={OperationalPointListData.map((op) => ({
+        ...op,
+        onClick: (id, ref) => {
+          handleWaypointClick(id, ref);
```

Comment:
As a user, it felt weird not to be able to close the menu, unless by clicking an actual action. This could be implemented directly in the application side, but it would be nice to have it directly in this example.

One simple way to handle it, is to give the `Menu` component a `onClickElsewhere` prop, that would allow you to catch when the users clicks anywhere else:

```tsx
type MenuProps = {
  menuRef: React.RefObject<HTMLDivElement>;
  items: MenuItem[];
  style?: React.CSSProperties;
  onClickElsewhere?: () => void;
};

const Menu = ({ menuRef, items, style, onClickElsewhere }: MenuProps) => {
  useEffect(() => {
    const handleClickBody = (e: MouseEvent) => {
      if (!menuRef.current || !onClickElsewhere) return;

      const node = e.target as Node;
      if (!menuRef.current.contains(node)) {
        onClickElsewhere();
      }
    };

    setTimeout(() => {
      document.body.addEventListener('click', handleClickBody);
    });
    return () => {
      document.body.removeEventListener('click', handleClickBody);
    };
  }, [menuRef, onClickElsewhere]);

  return (
    <div ref={menuRef} className="menu" style={style}>
      {items.map(({ title, icon, onClick }) => (
        <button key={title} type="button" className="menu-item" onClick={onClick}>
          <span className="icon">{icon}</span>
          <span>{title}</span>
        </button>
      ))}
    </div>
  );
};
```

Then you can close the menu when the user clicks somewhere else like this:

```tsx
<Menu
  menuRef={menuRef}
  items={menuItems}
  style={{ width: '305px', top: menuPosition.top, left: menuPosition.left }}
  onClickElsewhere={() => {
    setMenuPosition(null);
  }}
/>
```

Reply:
You're right, it's probably not necessary, the code is indeed simpler that way. As long as it's done "on the other side", it's all good to me :+1: